src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -1273,9 +1295,156 @@ export default {
 				this.showCheckout();
 			}
 		},
+		amountLeftOnLoan(loan) {
+			const loanFundraisingInfo = loan?.loanFundraisingInfo ?? { fundedAmount: 0, reservedAmount: 0 };
+			const { fundedAmount, reservedAmount } = loanFundraisingInfo;
+			return numeral(loan.loanAmount).subtract(fundedAmount).subtract(reservedAmount).value();
+		},
+		allocateLeftoverCredits(payload) {
+			const newPayload = payload;
+			let loanIdx = newPayload.loanIdx ? newPayload.loanIdx : 0;
+			// Allocate leftover credits to a loan that isn't already in the basket
+			// and make sure it needs at least the leftover amount we want to allocate
+			while (
+				this.itemsInBasket.indexOf(this.availableLoans.values[loanIdx].id) > -1
+				|| newPayload.lendAmount > this.amountLeftOnLoan(this.availableLoans.values[loanIdx])
+			) {
+				loanIdx += 1;
+				if (loanIdx === this.availableLoans.length - 1) {
+					break;
+				}
+			}
+			const loanId = this.availableLoans.values[loanIdx].id;
+			setLendAmount({
+				amount: newPayload.lendAmount,
+				apollo: this.apollo,
+				loanId
+			}).then(() => {
+				this.$emit('add-to-basket', { loanId, success: true });
+				this.leftoverCreditAllocationLoanId = loanId;
+				this.updateBasketState();
+			}).catch(e => {
+				logFormatter(`Failed to add loan with id ${loanId} to basket`, 'error');
+				let msg = 'There was a problem adding the loan to your basket';
+				switch (e[0].extensions.code) {
+					case 'reached_anonymous_basket_limit':
+						msg = e[0].message;
+						this.$showTipMsg(msg, 'error');
+						break;
+					case 'no_shares_added_regular_xb' || 'not_all_shared_added':
+						msg = e[0].message;
+						loanIdx += 1;
+						newPayload.loanIdx = loanIdx;
+						this.allocateLeftoverCredits(newPayload);
+						break;
+					default:
+						this.$showTipMsg(msg, 'error');
+				}
+			});
+		},
+		updateBasketItem(payload) {
```

Comment:
Could we please namespace this method "updateBasketItem", "completeRemoveBasketItem" and "trackItemRemoved" as they seem to be tightly tied to the new remaining credit allocation mechanisms.

Reply:
done